sympy/polys/matrices/sdm.py

```diff
@@ -970,3 +1012,34 @@ def sdm_particular_from_rref(A, ncols, pivots):
         if Ain is not None:
             P[j] = Ain / A[i][j]
     return P
+
+
+def sdm_applyfunc(M, shape, f, domain):
+    if domain.is_zero(f(domain.zero)):
+        return sdm_applyfunc_nonzero(M, f, domain)
+
+    nrows, ncols = shape
+    fzero = (f(domain.zero))
+    Mf = {i: {j: fzero for j in range(ncols)} for i in range(nrows)}
+
+    for i, Mi in M.items():
+        for j, Mij in Mi.items():
+                fMij = f(M[i][j])
+                if not domain.is_zero(fMij):
+                    Mf[i][j] = fMij
+                else:
+                    del Mf[i][j]
```

Comment:
You already have `Mij` so you don't need to look it up as `M[i][j]`. We can just use `if fMij` rather than `domain.is_zero`. That's what all of the other sparse routines do. It's probably worth doing `Mfi = Mf[i]` at the start of the outer loop. Also the indentation is wrong:
```
    for i, Mi in M.items():
        Mfi = Mf[i]
        for j, Mij in Mi.items():
            fMij = f(Mij)
            if fMij:
                Mfi[j] = fMij
            else:
                del Mfi[j]
```